lib/routes/selection_route.dart

```diff
@@ -16,28 +16,33 @@ class SelectionRoute extends StatefulWidget {
 class _SelectionRouteState extends State<SelectionRoute> {
   late final HomeRouter nestedHomeRouter = HomeRouter.selection(widget.selectionArguments);
   late final ContentSelectionController controller;
+  late bool settingsOpened;
 
   @override
   void initState() { 
     super.initState();
-    var settingsOpened = false;
+    settingsOpened = false;
     controller = ContentSelectionController.createAlwaysInSelection(
       context: context,
       actionsBuilder: (context) {
         final l10n = getl10n(context);
         final settingsPageBuilder = widget.selectionArguments.settingsPageBuilder;
         return [
-          if (settingsPageBuilder != null)
+          if (settingsPageBuilder != null && !settingsOpened)
```

Comment:
maybe add a `SelectionAppBar.defaultSelectionActionsBuilder` animation, using AnimatedSwitcher? also set similar curves and duration to what we have for other AnimatedSwitcher animations

Reply:
I'm not super familiar with the animation system, did you mean `AnimationSwitcher`?
Or is this what you had in mind?
```dart
if (settingsPageBuilder != null)
            AnimatedSwitcher(
              duration: const Duration(milliseconds: 240),
              switchInCurve: Curves.easeOutCubic,
              switchOutCurve: Curves.easeInCubic,
              transitionBuilder: SelectionAppBar.defaultSelectionActionsBuilder,
              child: settingsOpened
                ? const SizedBox.shrink()
                : NFIconButton(
                   ...
                ),
            ),
```

It doesn't show the config button when opening the selection route for the first time though. I would appreciate if you could give me some pointers.

Reply:
You are correct with your code snippet. The reason it doesn't work is actually a bug in defaultSelectionActionsBuilder, I pushed a fix to the nt4f04unds_widgets, please update the lock file to get this fix

Reply:
Actually, I don't like the looks of `SelectionAppBar.defaultSelectionActionsBuilder` here, let's just keep the defualt transitionBuilder (fade transition)